public/components/create_report/create_report.tsx

```diff
@@ -0,0 +1,310 @@
+    }
+  };
+
+  const ScheduleRadio = () => {
+    const radios = [
+      {
+        id: `${idPrefix}7`,
+        label: 'Now',
+      },
+      {
+        id: `${idPrefix}8`,
+        label: 'Future Date',
+      },
+      {
+        id: `${idPrefix}9`,
+        label: 'Recurring',
+      },
+    ];
+    return (
+      <EuiRadioGroup
+        options={radios}
+        idSelected={scheduleRadioIdSelected}
+        onChange={onChangeScheduleSettingsRadio}
+        name="scheduleRadioGroup"
+      />
+    );
+  };
+  
+  const timezone_options = [
+    { value: -4, text: 'EDT -04:00' },
+    { value: -5, text: 'CDT -05:00' },
+    { value: -6, text: 'MDT -06:00' },
+    { value: -7, text: 'MST/PDT -07:00' },
+    { value: -8, text: 'AKDT -08:00' },
+    { value: -10, text: 'HST -10:00' }
+  ];
+
+  const ScheduleFutureDatePicker = () => {  
+    const handleChangeScheduleDate = date => {
+      setScheduleFutureDate(date);
+    };
+
+    const onSelectOffsetChange = e => {
+      setScheduleUTCOffset(parseInt(e.target.value, 10));
+    };
+
+    return (
+      <div>
+        <EuiFormRow label="Time select on">
+          <EuiDatePicker
+            showTimeSelect
+            selected={scheduleFutureDate}
+            onChange={handleChangeScheduleDate}
+          />
+        </EuiFormRow>
+        <EuiFormRow label="UTC offset">
+          <EuiSelect
+            options={timezone_options}
+            value={scheduleUTCOffset}
+            onChange={onSelectOffsetChange}
+          />
+        </EuiFormRow>
+      </div>
+    );
+  }
+
+  const ScheduleRecurringDailyInput = () => {
+    const handleTimeChange = date => {
+      setScheduleRecurringStartDate(date);
+    };
+
+    const onSelectOffsetChange = e => {
+      setScheduleRecurringUTCOffset(parseInt(e.target.value, 10));
+    };
+
+    return (
+      <div>
+        <EuiFormRow label="Around">
+          <EuiDatePicker
+            showTimeSelect
+            showTimeSelectOnly
+            selected={scheduleRecurringStartDate}
+            onChange={handleTimeChange}
+            dateFormat="HH:mm"
+            timeFormat="HH:mm"
+          />
+        </EuiFormRow>
+        <EuiFormRow label="UTC offset">
+          <EuiSelect
+            options={timezone_options}
+            value={scheduleRecurringUTCOffset}
+            onChange={onSelectOffsetChange}
+          />
+        </EuiFormRow>
+      </div>
+    );
+  }
+
+  const ScheduleRecurringWeeklyInput = () => {
+    const recurringDayOptions = [
+      { value: 'Monday', text: 'Monday' },
+      { value: 'Tuesday', text: 'Tuesday' },
+      { value: 'Wednesday', text: 'Wednesday' },
+      { value: 'Thursday', text: 'Thursday' },
+      { value: 'Friday', text: 'Friday' },
+      { value: 'Saturday', text: 'Saturday' },
+      { value: 'Sunday', text: 'Sunday' },
+    ];
+
+    const onChangeDayOfWeek = (e) => {
+      setScheduleRecurringWeeklyDayOfWeek(e.target.value);
+    }
```

Comment:
`e: React.MouseEvent<>` missing type.

Reply:
Shouldnt this be `string` since the options are the days of the week which are strings?

Reply:
No the `e` should be of type event.